exercises/collatz-conjecture/test/collatz_conjecture_test.dart

```diff
@@ -0,0 +1,43 @@
+import 'package:collatz_conjecture/collatz_conjecture.dart';
+import 'package:test/test.dart';
+
+/// We are using a predicate to better match the error message from collatzConjecture.
+/// Knowing about predicates are not needed for completing this exercise,
+/// but you can find out more info at http://dartdoc.takyam.com/articles/dart-unit-tests/#matchers
+/// if you wish to learn more about predicates and matchers used in Unit Testing for Dart.
+
+void main() {
+  final collatzConjecture = new CollatzConjecture();
+  final onlyPositive =
+      predicate((ArgumentError e) => e is ArgumentError && e.message == 'Only positive numbers are allowed');
```

Comment:
Recommend giving it a custom description by add a second parameter `, 'an ArgumentError with the message "Only positive numbers are allowed"'`.

That way when the error fails it will say:

>  Expected: throws an ArgumentError with the message "Only positive numbers are allowed"

Which is much more descriptive than the default.